Шаблонизируй это

EgorViliavin opened the pull request:
Не трогал форму, есть некоторые моменты которые хочется уточнить.

---
:mortar_board: [Шаблонизируй это](https://up.htmlacademy.ru/ecmascript/23/user/2121439/tasks/6)

EgorViliavin committed:
Разбивает на компоненты
- Header (trip-info-view.js, trip-filter-view.js)
- Body (trip-form-sort-view.js, trip-list-view.js, trip-list-item-view.js, trip-form-view.js)
- На отдельные компоненты (trip-form-view.js, trip-list-item-view.js)
- компоненты находятся в папаках (form-elements, trip-list-event-element)

EgorViliavin committed:
Fix: Переустанавливает зависимости

EgorViliavin committed:
Fix: Переносит данные в const.js

EgorViliavin committed:
Fix: Создает папку mokky и переносит туда данные

EgorViliavin committed:
Fix: Переименовывает названия функций Element => Template

EgorViliavin committed:
Fix: Исправляет логику компонента trip-form-view.js

EgorViliavin committed:
Fix: исправляет слияние веток master & module1-task3

EgorViliavin committed:
Feat: Устанавливает prettier в проект
- добавляет .prettierrc.json в .gitignore

EgorViliavin committed:
Fix: Переносит данные в const.js

EgorViliavin committed:
Fix: Исправляет логику проверки типа формы

EgorViliavin committed:
Fix: Исправляет ошибки линтера

EgorViliavin committed:
Fix: Исправляет именование импорта

EgorViliavin committed:
Fix: Исправляет ошибки линтера

EgorViliavin committed:
Feat: Создает моковые данные точек
- Создает модуль trip-point-data.js в папке model
- Настраивает модуль на работу с данными
- Подключает модуль в main.js
- Настраивает presenter на прием данных

EgorViliavin committed:
Fix: Переименовывает model

EgorViliavin committed:
Feat: Настраиает компонент trip-list-event-element.js принимать данные

EgorViliavin committed:
Feat: Отрисовывает динамические данные точек

EgorViliavin committed:
Feat: Добавляет форматирование даты
- отображает динамическую дату в компоненте

EgorViliavin committed:
Fix: Исправляет ошибку зависимости eslint

EgorViliavin committed:
Fix: Исправляет ошибки по комментариям

EgorViliavin committed:
Fix: Исправляет ошибки  линтера

EgorViliavin committed:
Fix: Исправляет ошибку с работой кнопки

JuliaVolkova commented on src/mock/mocks.js:
```diff
@@ -0,0 +1,40 @@
+export const BUTTON_OFFERS = [
```
Некритично, но я бы сказала, что это не константа, это просто массив, поэтому его можно называть в `camelCase`.

JuliaVolkova commented on src/presenter/content-presenter.js:
```diff
@@ -0,0 +1,86 @@
+  constructor({ points, pointOffers, pointDestinations }) {
+    this.points = [...points];
+    this.pointOffers = [...pointOffers];
+    this.pointDestinations = [...pointDestinations];
```
Здесь я бы лучше сразу перешла на использование методов, которые описаны в модели - потому что дальше эти свойства  станут приватными, и возможность обратиться к ним извне будет только через публичные методы.

JuliaVolkova commented on src/view/trip-list-event-element/trip-list-event-element.js:
```diff
@@ -0,0 +1,116 @@
+        <div class="event__type">
+          <img class="event__type-icon" width="42" height="42" src="img/icons/${point.type.toLowerCase()}.png" alt="Event type icon">
+        </div>
+        <h3 class="event__title">${point.type} Amsterdam</h3>
```
_[ Nice to have ]_: можно использовать деструктуризацию из объекта `point`

JuliaVolkova commented on src/view/trip-list-event-element/trip-list-event-element.js:
```diff
@@ -0,0 +1,116 @@
+        timeToMilliseconds / MILLISECONDS
+      )}M`);
+    }
+  };
```
Я бы вынесла все эти манипуляции с форматированием даты и времени за пределы функции `createTripListEventTemplate` - сейчас она получается довольно сложной для чтения.

JuliaVolkova reviewed (commented):
Есть небольшие правки по коду, также важно реализовать пункт 6 из задания - научить форму редактирования тоже принимать данные от модели.

> Аналогичным образом перепишите View формы редактирования точки маршрута.

EgorViliavin committed:
Fix: Переделывает получение данных из модели в презентере

EgorViliavin committed:
Fix: Переименовывает названия моков

EgorViliavin committed:
Feat: Добавляет папку utils
- в ней создает файл point-time-formatter.js
- экспортирует его в trip-list-event-element.js
- разгружает функцию createTripListEventTemplate

EgorViliavin committed:
Feat: Добавляет новые модели в папку model
- additional-offer-model.js
- point-destinations-model.js
- убирает не нужные методы в points-model.js
- content-presenter принимает новые данные из моделей

JuliaVolkova commented on src/view/trip-list-event-element/create-event-offers.js:
```diff
@@ -0,0 +1,36 @@
+      if (offerItem.id === item) {
+        offers.push(offerItem);
+      }
+    });
```
Кажется, что здесь эта логика немного избыточна. Мы ведь знаем тип точки, чтобы быстро найти нужный массив офферов.

Как это можно было бы сделать в модели:
1. добавить там методы вот такого вида:

```
  getOffersByType(type) {
    const allOffers = this.getOffers();
    return allOffers.find((offer) => offer.type === type);
  }

  getOffersById(type, itemsId) {
    const offersType = this.getOffersByType(type);
    return offersType.offers.filter((item) => itemsId.find((id) => item.id === id));
  }
```
2. затем в презентере там, где мы инициализируем наши компоненты (точку или форму редактирования точки), мы обращаемся к этим методам.

Например, в твой метод `renderTripFormEditPointView` прокидывать не только `this.pointOffers`, но и тип точки - например,` offersType: this.pointsModel.getOffersByType(this.points[0].type),` - так при отрисовке офферов будет легче искать нужный массив офферов.
